source/IpcModule.ts

```diff
@@ -0,0 +1,57 @@
+import { ipcRenderer, ipcMain } from 'electron'
+
+type Promisified<T extends Function> = 
+    T extends (...args: infer P) => infer R
+        ? ( R extends Promise<any> 
+            ? (...args: P) => R
+            : (...args: P) => Promise<R>
+          )
+        : never
+    ;
+
+type Ipcfied<D> = {
+        [K in keyof D]: D[K] extends Function ? Promisified<D[K]> : never;
+    }
+
+export abstract class IpcModule {
+    public static forRenderer<T>(): Ipcfied<T> {
+        return new Proxy({}, {
+            get (target: any, prop: PropertyKey, receiver: any): any {
+                const methodName = prop.toString();
+                return async function(...args: any[]) {
+                    return ipcRenderer.invoke(methodName, args);
```

Comment:
Very good that you additionally switched to the sandboxing/invoking, as we'd need to fully sandbox the renderer either way